app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java

```diff
@@ -210,13 +214,35 @@ public void onBackPressed() {
                 BaseFragment.CURRENT_FRAGMENT_TAG
         );
         if (fragment instanceof AppBackClickTarget) {
-            ((AppBackClickTarget) fragment).appBackClick();
+            waitForDoubleBackPressed();
         }
         for (Fragment fragmentInApp : getSupportFragmentManager().getFragments()) {
             if (fragmentInApp instanceof AppResumeTarget) {
                 ((AppResumeTarget) fragmentInApp).appResume();
             }
         }
+
+    }
+
+    private void waitForDoubleBackPressed() {
+        if (doubleBackToExitPressedOnce) {
```

Comment:
What for are you starting new activity instance after single back click?

Reply:
i read somewhere that in order to save app state should make a new activity

Reply:
do you have a better suggestion?

Reply:
You should just show toast. If you re-run an activity, the state changing.

Reply:
I made some changes, i think it's fine now

Reply:
You don't need to start a new activity. Just show the dialog for user. If you start new activity, all user's data is erased.

Reply:
where am I making a new activity????
are u sure you are checking my last commit?
i do NOT make a  new activity in my last comit

Reply:
In your last commit all fine but can you move this implementation to `HashCalculatorFragment`? This needs because `HashCalculatorFragment` is a "main" fragment of the app and user can exit by double click just only from it. Thanks.